2216/2/bad_words_analyzer.rb

```diff
@@ -0,0 +1,38 @@
+class BadWordsAnalyzer
+  def initialize(battle)
+    @battle = battle
+  end
+
+  def analyze_bad
+    words = divide_into_words
+    words_count = words.size
+    bad_words_count = BadWordsAnalyzer.count_bad_words(words)
```

Comment:
This is confusing that you're calling `BadWordsAnalyzer` class method from `BadWordsAnalyzer` object!

If you want to proceed with `BadWordsAnalyzer` class, I'd expect to see the following interface
```
BadWordsAnalyzer.new(battle).analyze # => returns data structure
```